examples/kubernetes/llamacpp/charts/embedding/templates/deployment.yaml

```diff
@@ -0,0 +1,161 @@
+{{ $port := .Values.service.port }}
+{{- range $modelName, $modelConfig := .Values.models }}
+{{- if $modelConfig.enabled }}
+
+{{- if $modelConfig.download }}
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "embedding.fullname" $ | lower }}-download-{{ $modelName }}
+  labels:
+    app: {{ include "embedding.fullname" $ | lower }}
+spec:
+  template:
+    spec:
+      initContainers:
+        - name: fix-permissions
+          image: busybox
+          command: ["sh", "-c", "chmod -R 777 /models"]
+          volumeMounts:
+            - name: {{ include "embedding.fullname" $ | lower }}-pv-{{ $modelName }}
+              mountPath: /models
+        
+      containers:
+      - name: init-job
+        image: alpine/k8s:1.27.11
+        command:
+          - sh
+          - -c
+          - |
+            set -e
+            if curl -L {{ $modelConfig.url }} --output /models/{{ $modelName }}/{{ $modelName }}.gguf; then
```

Comment:
It will not support sharded model files. Better to let llama.cpp server handles the initial download

Reply:
Ok but then we wont be able to have a job running it. This will prevent us from updating it using kubectl apply. Also i dont believe llama server supports autodownload? I know Ollama does. When llamacpp server container tries to start it needs a model file to point to or else it errors out.

Reply:
No I developed that feature some time ago, see the doc.